worker/meson.build

```diff
@@ -0,0 +1,227 @@
+  'src/RTC/IceCandidate.cpp',
+  'src/RTC/IceServer.cpp',
+  'src/RTC/KeyFrameRequestManager.cpp',
+  'src/RTC/NackGenerator.cpp',
+  'src/RTC/PipeConsumer.cpp',
+  'src/RTC/PipeTransport.cpp',
+  'src/RTC/PlainTransport.cpp',
+  'src/RTC/PortManager.cpp',
+  'src/RTC/Producer.cpp',
+  'src/RTC/RateCalculator.cpp',
+  'src/RTC/Router.cpp',
+  'src/RTC/RtpListener.cpp',
+  'src/RTC/RtpObserver.cpp',
+  'src/RTC/RtpPacket.cpp',
+  'src/RTC/RtpProbationGenerator.cpp',
+  'src/RTC/RtpStream.cpp',
+  'src/RTC/RtpStreamRecv.cpp',
+  'src/RTC/RtpStreamSend.cpp',
+  'src/RTC/RtxStream.cpp',
+  'src/RTC/SctpAssociation.cpp',
+  'src/RTC/SctpListener.cpp',
+  'src/RTC/SenderBandwidthEstimator.cpp',
+  'src/RTC/SeqManager.cpp',
+  'src/RTC/SimpleConsumer.cpp',
+  'src/RTC/SimulcastConsumer.cpp',
+  'src/RTC/SrtpSession.cpp',
+  'src/RTC/StunPacket.cpp',
+  'src/RTC/SvcConsumer.cpp',
+  'src/RTC/TcpConnection.cpp',
+  'src/RTC/TcpServer.cpp',
+  'src/RTC/Transport.cpp',
+  'src/RTC/TransportCongestionControlClient.cpp',
+  'src/RTC/TransportCongestionControlServer.cpp',
+  'src/RTC/TransportTuple.cpp',
+  'src/RTC/TrendCalculator.cpp',
+  'src/RTC/UdpSocket.cpp',
+  'src/RTC/WebRtcTransport.cpp',
+  'src/RTC/Codecs/H264.cpp',
+  'src/RTC/Codecs/VP8.cpp',
+  'src/RTC/Codecs/VP9.cpp',
+  'src/RTC/RtpDictionaries/Media.cpp',
+  'src/RTC/RtpDictionaries/Parameters.cpp',
+  'src/RTC/RtpDictionaries/RtcpFeedback.cpp',
+  'src/RTC/RtpDictionaries/RtcpParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpCodecMimeType.cpp',
+  'src/RTC/RtpDictionaries/RtpCodecParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpEncodingParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpHeaderExtensionParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpHeaderExtensionUri.cpp',
+  'src/RTC/RtpDictionaries/RtpParameters.cpp',
+  'src/RTC/RtpDictionaries/RtpRtxParameters.cpp',
+  'src/RTC/SctpDictionaries/SctpStreamParameters.cpp',
+  'src/RTC/RTCP/Packet.cpp',
+  'src/RTC/RTCP/CompoundPacket.cpp',
+  'src/RTC/RTCP/SenderReport.cpp',
+  'src/RTC/RTCP/ReceiverReport.cpp',
+  'src/RTC/RTCP/Sdes.cpp',
+  'src/RTC/RTCP/Bye.cpp',
+  'src/RTC/RTCP/Feedback.cpp',
+  'src/RTC/RTCP/FeedbackPs.cpp',
+  'src/RTC/RTCP/FeedbackRtp.cpp',
+  'src/RTC/RTCP/FeedbackRtpNack.cpp',
+  'src/RTC/RTCP/FeedbackRtpTmmb.cpp',
+  'src/RTC/RTCP/FeedbackRtpSrReq.cpp',
+  'src/RTC/RTCP/FeedbackRtpTllei.cpp',
+  'src/RTC/RTCP/FeedbackRtpEcn.cpp',
+  'src/RTC/RTCP/FeedbackRtpTransport.cpp',
+  'src/RTC/RTCP/FeedbackPsPli.cpp',
+  'src/RTC/RTCP/FeedbackPsSli.cpp',
+  'src/RTC/RTCP/FeedbackPsRpsi.cpp',
+  'src/RTC/RTCP/FeedbackPsFir.cpp',
+  'src/RTC/RTCP/FeedbackPsTst.cpp',
+  'src/RTC/RTCP/FeedbackPsVbcm.cpp',
+  'src/RTC/RTCP/FeedbackPsLei.cpp',
+  'src/RTC/RTCP/FeedbackPsAfb.cpp',
+  'src/RTC/RTCP/FeedbackPsRemb.cpp',
+  'src/RTC/RTCP/XR.cpp',
+  'src/RTC/RTCP/XrDelaySinceLastRr.cpp',
+  'src/RTC/RTCP/XrReceiverReferenceTime.cpp',
+]
+
+cpp = meson.get_compiler('cpp')
+
+openssl_dep = dependency(
+  'openssl',
```

Comment:
Meson has an experimental module for building wraps using configure/make style build systems: https://mesonbuild.com/External-Project-module.html

It will probably work with openssl, assuming you build statically (which third-party vendored dependencies should do anyway regardless of build system. :D)

Reply:
Hm, I might try that. I experimented with https://github.com/viaduck/openssl-cmake with CMake integration, but it didn't build anything despite not throwing errors.

On Linux it might be fine to use configure/make, but at the same time it is kind of a pain to set a lot of components (like Perl for OpenSSL) on Windows. Having just Meson+Ninja+C compiler is much nicer.

I might end up writing meson.build for OpenSSL, I'll see what maintainers think about Meson in general first.

Reply:
Right, it's experimental after all. :D (And a teeny bit of a hack.)